cookbooks/04_Cookbook.ipynb

```diff
@@ -0,0 +1,449 @@
+    "          \n",
+    "          Current weather condition at \"{{_[0]}}\":\n",
+    "          \"{{_[1]}}\"\n",
+    "\n",
+    "          Top tourist attractions in \"{{_[0]}}\":\n",
+    "          \"{{_[2]}}\"\n",
+    "\n",
+    "          Suggest outdoor or indoor activities based on the above information.\n",
+    "    unwrap: true\n",
+    "                   \n",
+    "\"\"\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Creating/Updating a task to generate a sarcastic response to a given text using a Intergation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# creating the task object\n",
+    "task = client.tasks.create_or_update(\n",
+    "    task_id=TASK_UUID,\n",
+    "    agent_id=AGENT_UUID,\n",
+    "    **task_def\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#listing all tasks for a given agent\n",
+    "get_tasks = client.tasks.list(agent_id=AGENT_UUID)\n",
+    "get_tasks.items "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Creating an Execution\n",
+    "\n",
+    "An execution is a single run of a task. It is a way to run a task with a specific set of inputs."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Creates a execution worflow for the Task defined in the yaml file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "execution = client.executions.create(\n",
+    "    task_id=task.id,\n",
+    "    input={\n",
+    "        \"locations\": [\"New York\"]\n",
+    "    }\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "execution.id"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# getting the execution details\n",
+    "execution = client.executions.get(execution.id)\n",
+    "#printing the output\n",
+    "execution.output"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Streaming execution steps of the defined Task."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 31,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "client.executions.transitions.stream(execution_id=execution.id)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Listing the execution states."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
```

Comment:
mention that list transitions doesnt wait, just lists whatever transitions have happened so far

Reply:
and move `.list()` above `.stream()`